test/box-luatest/gh_8587_func_index_unique_violation_test.lua

```diff
@@ -0,0 +1,106 @@
+local t = require('luatest')
```

Comment:
It would be nice to have a unit test for functional tuple comparator with nullable key as well. We already have all needed infrastructure for this in `unit/key_def.c`, so it shouldn't be difficult.

Reply:
Done.